### PR TITLE
Aztec decoder returns raw bytes

### PR DESCRIPTION
**Pull request by tanelihuuskonen**
The Aztec decoder now packs the code bits into bytes and returns them as "raw bytes", even though the error correction codes and stuffing have been already removed.  Also added a couple of test cases for the byte-packing function.  This fixes issue #381.

I hereby grant everyone the right to use and distribute this contribution under the same terms as the original program.

### CODE REVIEW

File: core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java [hunk 1]
```diff
@@ -332,6 +333,28 @@ private static int readCode(boolean[] rawbits, int startIndex, int length) {
...
+    if (n >= 8) {
+      return (byte) readCode(rawbits, startIndex, 8);
+    }
+    return (byte) (readCode(rawbits, startIndex, n) << (8 - n));
+  }
+
+  /**
+   * Packs a bit array into bytes, most significant bit first
+   */
+  public static byte[] convertBoolArrayToByteArray(boolean[] boolArr) {
```
Comment (srowen) [inline core/src/main/java/com/google/zxing/aztec/decoder/Decoder.java]: Can be package-private?

File: core/src/test/java/com/google/zxing/aztec/decoder/DecoderTest.java [hunk 2]
```diff
@@ -105,4 +106,35 @@ public void testDecodeTooManyErrors2() throws FormatException {
...
+  private static void assertEqualByteArrays(byte[] b1, byte[] b2) {
+    assertEquals(b1.length, b2.length);
+    for (int i = 0; i < b1.length; i++) {
+      assertEquals(b1[i], b2[i]);
+    }
+  }
+
+  @Test
+  public void testRawBytes() {
+    boolean bool0[] = {};
```
Comment (srowen) [inline core/src/test/java/com/google/zxing/aztec/decoder/DecoderTest.java]: Nit: `boolean[] bool0...` (Java-style array declarations)

File: core/src/test/java/com/google/zxing/aztec/decoder/DecoderTest.java [hunk 3]
```diff
@@ -105,4 +106,35 @@ public void testDecodeTooManyErrors2() throws FormatException {
     new Decoder().decode(r);
   }
 
+  private static void assertEqualByteArrays(byte[] b1, byte[] b2) {
```
Comment (srowen) [inline core/src/test/java/com/google/zxing/aztec/decoder/DecoderTest.java]: JUnit already has ... http://junit.sourceforge.net/javadoc/org/junit/Assert.html#assertArrayEquals(byte[], byte[])

**Comment by tanelihuuskonen** [comment]
Thanks for the comments.  I only started learning Java a couple of months ago, so my C habits keep showing up.  I just committed my corrections.

**Pull request merged**